kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeBlock.kt

```diff
@@ -361,9 +363,25 @@ public class CodeBlock private constructor(
       else -> throw IllegalArgumentException("expected name but was $o")
     }
 
-    private fun argToLiteral(o: Any?) = o
+    private fun argToLiteral(o: Any?) = if (o is Number) formatNumericValue(o) else o
 
-    private fun argToString(o: Any?) = o?.toString()
+    private fun argToString(o: Any?) = if (o is Number) formatNumericValue(o) else o?.toString()
```

Comment:
Please revert. We should not alter number formatting flowing into string literals.

Reply:
I reverted this string formatting